function_migrations.go

```diff
@@ -203,6 +204,137 @@ func migrateToSpecVersion(f Function, m migration) (Function, error) {
 	return f, nil
 }
 
+// This migration makes sure use the 1.0.0 format. To avoid unmarshalling issues with the old format
+// this migration needs to be executed first. Further migrations will operate on this new struct
+func migrateTo100Structure(f1 Function, m migration) (Function, error) {
+	// Load the Function using pertinent parts of the previous version's schema:
+	f0Filename := filepath.Join(f1.Root, FunctionFile)
```

Comment:
@lkingland I am also experiencing some issues with this line. As I assumed that migrations are additive, but this line basically reload the file for every migration that we define, hence the latest migration will, in some way, ignore all the previous ones, because it is loading the file from the file system instead of the function that is returned from the previous migration. 

The test that is currently failing (`migrateToBuilderImages`) show cases this issue. Advice is appreciated here.. as I might be missing some details or design decisions.

Reply:
yes.. a migrated.Write() will be the right thing to do.. let me try with that and see how that goes.

Reply:
ok adding `migrated.Write()` opens a whole new can of worms :) The problem with that approach is that after running the first migration, the file moves to the latest format, and without a field by field migration on the first one executed, we will loose data.
Also this makes all the test to modify the files, which will require copying the files to a temp location to then do the migrations. 

I will try the second approach which is doing the additive migrations on a memory structure that is written only at the end. I am guessing that was the original approach based on how the test are working today.

Reply:
For now, I've fixed the issue to make sure that the migration is really additive.. this basically means that I am not replacing values that were modified by previous versions. For example in arrays or maps I am just bringing the old values forward, without destroying values that might be in the new structure already.